File: pset1/mario_hard.c
#include <stdio.h>
#include <cs50.h>

int main(void) {
  // initialize variables
  int height;

  // prompt user for height
  printf("Height: ");
  do {
    height = GetInt();
    if (height < 0 || height > 23)
      printf("Retry: ");
  } while (height < 0 || height > 23);

  // print the bypramid
  for (int i = 0; i < height; i++) {
    for (int j = 1; j < (height - i); j++)
      printf(" ");
    for (int k = 0; k < i; k++)
      printf("#");
    printf("#  ");
    for (int l = 0; l < i; l++)
      printf("#");
    printf("#\n");
  }

  // render success
  return 0;
}

File: pset6/mario_hard.py
from __future__ import print_function
import cs50

# print the pyramid
def main():
    # recover height from get_positive_int() function
    height = get_positive_int()

    # print the pyramid
    for n in range(height):
        for i in range(height - (n + 1)):
            print(" ", end="")

        for j in range(n + 1):
            print("#", end="")

        print("  ", end="")

        for k in range(n):
            print("#", end="")

        print("#")

# prompt the user height of pyramid
def get_positive_int():
    print("Height: ", end="")

    # prompt the user for a positive integer
    while True:
        height = cs50.get_int()

        # case for out of bound value
        if height > 23 or height < 0:
            print("Retry: ", end="")

        # case for in bound value
        if 0 <= height <= 23:
            break

    return height

# call main
if __name__ == "__main__":
    main()
